test/snippet/core/metafunction/deferred_crtp_base.cpp

```diff
@@ -0,0 +1,49 @@
+//! [code]
+#include <string>
+#include <type_traits>
+
+#include <seqan3/core/metafunction/deferred_crtp_base.hpp>
+
+// Defines a crtp_base class with an additional value type.
+template <typename derived_t, typename value_t>
+class base1
+{
+public:
+
+    value_t func1() const
+    {
+        return {"instance of base1"};
+    }
+};
+
+// Defines a crtp_base class with an additional return type and a parameter_t.
```

Comment:
```suggestion
// Defines a crtp_base class with an additional value type and a parameter_t.
```
?

Reply:
... and a parameter type. (in a comment we don't need to write parameter_t)